File: 1813-maximum-erasure-value/maximum-erasure-value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        Set set = new HashSet<Integer>();
        int result = 0, currSum = 0, left = 0, right = 0;
        while(right < nums.length){
            while(set.contains(nums[right])){
                currSum-=nums[left];
                set.remove(nums[left++]);
            }
            currSum+=nums[right];
            set.add(nums[right++]);
            result = Math.max(currSum,result);
        }
        return result;
    }
}

File: 1813-maximum-erasure-value/maximum-erasure-value.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumUniqueSubarray = function(nums) {
    const set = new Set();
    let left=0,right=0,res=0,currSum=0;
    while(right<nums.length){
        while(set.has(nums[right])){
            currSum-=nums[left];
            set.delete(nums[left++]);
        }
        currSum+=nums[right];
        set.add(nums[right++]);
        res=Math.max(res,currSum);
    }
    return res;
};